rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,606 @@
+> Network health checks can be used to ensure that the deployment's network settings are correct before the user moves on from the deployment step.
+> The other failure modes do not depend on the deployment settings, so it makes sense to keep those tests in a separate subsequent step.
+
+#### User story: web UI resource health indicators
+
+Alice has deployed a Teleport database agent and enrolled a PostgreSQL database with health checks enabled and using the default settings.
+
+Some time later, another user reconfigures the database firewall rules to restrict the allowed inbound traffic IP ranges, but they inadvertently block the Teleport database agent from reaching the database over the network.
+
+Alice logs into the web UI and navigates to the resources page. 
+
+Alice sees a tooltip or warning icon on the database she set up earlier.
+
+Alice clicks on the tooltip/icon to see the message:
+
+    The Teleport database service proxying access to this database cannot reach the database endpoint.
+    <link to health check or network troubleshooting doc>
+
+    Affected Teleport database service:
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Error: "(tcp) failed: Operation timed out"
+
+Among other suggestions, the documentation includes:
+- check that the database is actually listening on `<port>`
+- check that database inbound TCP traffic is allowed from the agent to `<ip : port>`
+- check that agent outbound TCP traffic is allowed to `<ip : port>`
+
+After checking the linked documentation and some investigation Alice diagnoses that the timeout is caused by the database's firewall dropping packets from the agent's IP, which she resolves by updating the firewall rules.
+
+After a short (<20s) time the threshold is reached and the agent changes its status to `healthy`.
+
+The warning on the resource tile goes away.
+
+Alice checks the health status manually with tctl:
+
+```
+$ tctl get db_server/example
+
+kind: db_server
+metadata:
+  expires: "2025-02-21T02:03:47.398926Z"
+  name: example
+  revision: 43e96231-faaf-43c3-b9b8-15cf91813389
+spec:
+  ...*snip*...
+  host_id: 278be63c-c87e-4d7e-a286-86002c7c45c3
+  hostname: mars.internal
+  target_health:
+    addr: example.com:5432
+    protocol: TCP
+    transition_timestamp: "2025-02-19T01:53:39.144218Z"
+    transition_reason: "healthy threshold reached"
+    status: healthy
+  version: 18.0.0-dev
+version: v3
+```
+
+#### Out of scope
+
+The following ideas were considered, but ultimately cut from the initial implementation of this RFD to reduce its scope:
+
+1. Add a new type of notification routing rule that the auth service can use to send a notification when a target becomes unhealthy, much like the `access_request_routing_rules` added in [RFD 87 - Access request notification routing](./0087-access-request-notification-routing.md).
+2. Automatically create a "user task" in the AWS integration dashboard when an AWS RDS database's health status is unhealthy.
```

Comment:
The notification routing rule I think might be a little too much but user task is a good one - we should definitely do it. Ok to do as a follow up.